projectional/src/main/java/jetbrains/jetpad/projectional/cell/BaseProjectionalSynchronizer.java

```diff
@@ -275,15 +276,29 @@ public void addMapperProcessor(MapperProcessor<SourceItemT, Cell> processor) {
 
   @Override
   public void setOnLastItemDeleted(Runnable action) {
+    if (!myPlaceholderEnabled && action == null) {
+      throw new IllegalArgumentException("Can't set the default onLastItemDeleted action when the placeholder is disabled");
+    }
     myOnLastItemDeleted = action;
   }
 
   @Override
   public void setPlaceholderText(String text) {
+    if (!myPlaceholderEnabled) {
+      throw new IllegalStateException("The placeholder is disabled");
+    }
     myPlaceholderText = text;
   }
 
   @Override
+  public void disablePlaceholder() {
+    if (myOnLastItemDeleted == null) {
+      throw new IllegalStateException("Please provide onLastItemDeleted action first by calling setOnLastItemDeleted()");
```

Comment:
"disabling a placeholder removes onLastItemDeleted handler". 
How? If you have a handler it will be there.

"This causes unexpected focus lost"
Then specify the handler. I can see the point here. But it's trading something that can be fixed and focus lost (the program doesn't die) for the documentation/configuration IllegalStateException that will kill the program. I can understand it but isn't it a little extreme? Maybe a warning at least?

"NPE is also possible"

How NPE is possible in this code?
`TextCell placeHolder = myTargetCellList.getPlaceHolder(); if (placeHolder != null) { placeHolder.focus(); } `